components/addContact/index.jsx

```diff
@@ -67,28 +70,28 @@ export function handleSubmit({
     setIsLoading(true);
 
     try {
-      const { name, webId, types } = await fetchProfile(iri, fetch);
+      const { names, webId, types } = await getFullProfile(iri, session);
       const existingContact = await findContactInAddressBook(
         people,
         webId,
         fetch
       );
-
       if (existingContact) {
         alertError(EXISTING_WEBID_ERROR_MESSAGE);
         setIsLoading(false);
         return;
       }
-
-      const contact = { webId, fn: name || null };
+      const contact = {
+        webId,
+        fn: names[0],
```

Comment:
I assume `names[0]` is correct and that we wouldn't want to do a `first`/`find` operation to select the first name that has a value?

Reply:
no null/undefined values are put into that array

Reply:
okay! I was thinking more if there was a profile you could read or the first profile didn't have a name or some such. Though I'm not super familiar with the code, hence a comment asking :)